lib/attachments.js

```diff
@@ -2,71 +2,57 @@ const client = require('elasticio-rest-node')();
 const http = require('http');
 const url = require('url');
 
-exports.addAttachment = addAttachment;
-
-function addAttachment(msg, name, stream, contentLength) {
-    console.log("About to upload attachment stream of length %s", contentLength);
-
-    return client.resources.storage
-        .createSignedUrl()
-        .then(onSignedUrl);
-
-    function onSignedUrl(result) {
-        console.log("Created signed URL=%s", result.get_url);
-
-        const opts = createRequestOptions(contentLength, result.put_url);
-
-        return uploadFile(stream, opts)
-            .then(addUrlAttachment.bind(null, msg, name, result.get_url, contentLength));
-    }
-}
-
-function addUrlAttachment(msg, name, url, size) {
-
-    msg.attachments[name] = {
-        url: url,
-        size: size
-    };
+function createRequestOptions(contentLength, putUrl) {
+  const opts = url.parse(putUrl);
 
-    return Promise.resolve();
+  opts.method = 'PUT';
+  opts.headers = {
+    'Content-Length': contentLength,
+  };
+  return opts;
 }
 
-
 function uploadFile(stream, options) {
```

Comment:
Maybe, have the sense use AttachmentProcessofor downloading/uploading attachments.